include/fiction/algorithms/verification/virtual_miter.hpp

```diff
@@ -0,0 +1,121 @@
+#include <mockturtle/traits.hpp>
+
+#include <algorithm>
+#include <optional>
+
+namespace fiction
+{
+
+/*! \brief Creates a combinational miter from two networks.
+ *
+ * This method combines two networks that have the same number of primary
+ * inputs and the same number of primary outputs into a miter.  The miter
+ * has the same number of inputs and one primary output.  This output is the
+ * OR of XORs of all primary output pairs.  In other words, the miter outputs
+ * 1 for all input assignments in which the two input networks differ.
+ *
+ * All networks may have different types.  The method returns an optional, which
+ * is `nullopt`, whenever the two input networks don't match in their number of
+ * primary inputs and primary outputs.
+ */
+template <class NtkDest, class NtkSource1, class NtkSource2>
+std::optional<NtkDest> miter(NtkSource1 const& ntk1, NtkSource2 const& ntk2)
+{
+    static_assert(mockturtle::is_network_type_v<NtkSource1>, "NtkSource1 is not a network type");
+    static_assert(mockturtle::is_network_type_v<NtkSource2>, "NtkSource2 is not a network type");
+    static_assert(mockturtle::is_network_type_v<NtkDest>, "NtkDest is not a network type");
+
+    static_assert(mockturtle::has_num_pis_v<NtkSource1>, "NtkSource1 does not implement the num_pis method");
+    static_assert(mockturtle::has_num_pos_v<NtkSource1>, "NtkSource1 does not implement the num_pos method");
+    static_assert(mockturtle::has_num_pis_v<NtkSource2>, "NtkSource2 does not implement the num_pis method");
+    static_assert(mockturtle::has_num_pos_v<NtkSource2>, "NtkSource2 does not implement the num_pos method");
+    static_assert(mockturtle::has_create_pi_v<NtkDest>, "NtkDest does not implement the create_pi method");
+    static_assert(mockturtle::has_create_po_v<NtkDest>, "NtkDest does not implement the create_po method");
+    static_assert(mockturtle::has_create_xor_v<NtkDest>, "NtkDest does not implement the create_xor method");
+    static_assert(mockturtle::has_create_nary_or_v<NtkDest>, "NtkDest does not implement the create_nary_or method");
+
+    auto num_pis1 = ntk1.num_pis();
+    auto num_pis2 = ntk2.num_pis();
+    if constexpr (has_num_real_pis_v<NtkSource1>)
+    {
+        num_pis1 = ntk1.num_real_pis();
+    }
+    if constexpr (has_num_real_pis_v<NtkSource2>)
+    {
+        num_pis2 = ntk2.num_real_pis();
+    }
+    /* both networks must have same number of inputs and outputs */
+    if ((num_pis1 != num_pis2) || (ntk1.num_pos() != ntk2.num_pos()))
+    {
+        return std::nullopt;
+    }
+
+    /* create primary inputs */
+    NtkDest                                  dest;
+    std::vector<mockturtle::signal<NtkDest>> pis;
+    for (auto i = 0u; i < num_pis1; ++i)
+    {
+        pis.push_back(dest.create_pi());
+    }
+
+    std::shared_ptr<NtkSource1> ntk1_copy;
+    if constexpr (has_remove_virtual_input_nodes_v<NtkSource1>)
+    {
+        ntk1_copy = std::make_shared<NtkSource1>(ntk1);
```

Comment:
This is not a deep copy. I believe the networks have a `clone()` (?) function for that